src/rtk/features/reactions/postReactionsSlice.ts

```diff
@@ -0,0 +1,155 @@
+import { createAsyncThunk, createEntityAdapter, createSlice } from '@reduxjs/toolkit'
+import { getFirstOrUndefined, isDef } from '@subsocial/utils'
+import { createSelectUnknownIds, FetchManyArgs, ThunkApiConfig } from 'src/rtk/app/helpers'
+import { SelectManyFn } from 'src/rtk/app/hooksCommon'
+import { RootState } from 'src/rtk/app/rootReducer'
+import { AccountId, PostId } from 'src/types'
+import { idToPostId } from 'src/types/utils'
+
+export enum ReactionEnum {
+  Upvote = 'Upvote',
+  Downvote = 'Downvote'
+}
+
+export type ReactionId = string
+export type ReactionType = 'Upvote' | 'Downvote'
+
+export type Reaction = {
+  reactionId?: ReactionId
+  kind?: ReactionType
+}
+
+export type ReactionStruct = Reaction & {
+  id: PostId
+}
+
+const adapter = createEntityAdapter<ReactionStruct>()
+
+const selectors = adapter.getSelectors<RootState>(state => state.postReactions)
+
+// Rename the exports for readability in component usage
+export const {
+  selectById: selectReactionByPostId,
+  selectIds: selectPostIds,
+  // selectEntities: selectFollowedSpaceIdsEntities,
+  // selectAll: selectAllFollowedSpaceIds,
+  // selectTotal: selectTotalSpaceFollowers
+} = selectors
+
+export const selectPostReactionsByPostIds:
+  SelectManyFn<{}, ReactionStruct> = (
+    state, 
+    { ids }
+  ) => {
+    const reactions: ReactionStruct[] = []
+  
+    ids.forEach(id => {
+      const reaction = selectReactionByPostId(state, id)
+
+      if (reaction) {
+        reactions.push(reaction)
+      }
+    })
+
+    return reactions
+  }
+
+export const selectPostReactionByPostId = (state: RootState, id: PostId) => 
+  getFirstOrUndefined(selectPostReactionsByPostIds(state, { ids: [ id ] }))
+
+type Args = {
+  myAddress?: AccountId,
+  reload?: boolean
+}
+
+export type FetchManyReactionsArgs = FetchManyArgs<Args>
+
+export type FetchManyRes = ReactionStruct[]
+
+export const selectUnknownPostIds = createSelectUnknownIds(selectPostIds)
+
+export const fetchPostReactions = createAsyncThunk
+  <FetchManyRes, FetchManyReactionsArgs, ThunkApiConfig>(
+  'postReactionByAccount/fetchMany',
+  async ({ api, ids, myAddress, reload }, { getState }): Promise<FetchManyRes> => {
+ 
+    let newIds = ids as string[]
+
+    if (!reload || !myAddress) {
+      newIds = selectUnknownPostIds(getState(), ids)
+      if (!newIds.length || !myAddress) {
```

Comment:
This code looks bad with `!myAddress` used two times.
This arg should be required and use this code on top of this fetch func:
```
if (!myAddress) return []

```